lib/facter/gluster.rb

```diff
@@ -24,33 +26,29 @@
     end
   end
   output = Facter::Util::Resolution.exec("#{binary} peer status")
-  peer_count = $1.to_i if output =~ /^Number of Peers: (\d+)$/
-    if peer_count > 0 then
-      peer_list = output.scan(/^Hostname: (.+)$/).flatten.join(',')
-      # note the stderr redirection here
-      # `gluster volume list` spits to stderr :(
-      output = Facter::Util::Resolution.exec("#{binary} volume list 2>&1")
-      if output != "No volumes present in cluster" then
-        output.split.each do |vol|
-          info = Facter::Util::Resolution.exec("#{binary} volume info #{vol}")
-          vol_status = $1 if info =~ /^Status: (.+)$/
-            bricks = info.scan(/^Brick[^:]+: (.+)$/).flatten
-          volume_bricks[vol] = bricks
-          options = info.scan(/^(\w+\.[^:]+: .+)$/).flatten
-          if options then
-            volume_options[vol] = options
-          end
-          if vol_status == "Started" then
-            # if `gluster volume status` fails for some reason, it spits to stderr,
-            # so we suppress that.
-            status = Facter::Util::Resolution.exec("#{binary} volume status #{vol} 2>/dev/null")
-            if status =~ /^Brick/ then
-              volume_ports[vol] = status.scan(/^Brick [^\t]+\t+(\d+)/).flatten.uniq.sort
-            end
-          end
+  peer_count = Regexp.last_match[1].to_i if output =~ /^Number of Peers: (\d+)$/
+  if peer_count > 0
+    peer_list = output.scan(/^Hostname: (.+)$/).flatten.join(',')
+    # note the stderr redirection here
+    # `gluster volume list` spits to stderr :(
+    output = Facter::Util::Resolution.exec("#{binary} volume list 2>&1")
+    if output != 'No volumes present in cluster'
+      output.split.each do |vol|
+        info = Facter::Util::Resolution.exec("#{binary} volume info #{vol}")
+        # rubocop:disable Metrics/BlockNesting
```

Comment:
There are a bunch of rubocop nesting violations throughout, have decided to ignore them for now. (This file is a good candidate for refactoring)